source/compiler-core/slang-json-parser.cpp

```diff
@@ -0,0 +1,384 @@
+// slang-json-parser.cpp
+#include "slang-json-parser.h"
+
+#include "slang-json-diagnostics.h"
+
+/*
+https://www.json.org/json-en.html
+*/
+
+namespace Slang {
+
+SlangResult JSONParser::_parseObject()
+{
+    SLANG_RETURN_ON_FAIL(m_lexer->expect(JSONTokenType::LBrace));
```

Comment:
Eventually we may want this code to support a "recovery" system closer to what the actual Slang parser does, as a QOI feature.